controllers/watcher_controller.go

```diff
@@ -683,6 +691,35 @@ func (r *WatcherReconciler) ensureDBSync(
 	return ctrlResult, nil
 }
 
+func (r *WatcherReconciler) createTopLevelSecret(
+	ctx context.Context,
+	helper *helper.Helper,
+	instance *watcherv1beta1.Watcher,
+	transportURLSecret corev1.Secret,
+	inputSecret corev1.Secret,
+) (string, error) {
+	data := map[string]string{
+		instance.Spec.PasswordSelectors.Service: string(inputSecret.Data[instance.Spec.PasswordSelectors.Service]),
+		TransportURLSelector:                    string(transportURLSecret.Data[TransportURLSelector]),
+	}
```

Comment:
Should we also add some form of credentials for the database too? at least the name of the database secret, so that we don't need to guess it from subCRs.

Reply:
ack, let me give it a try and will update the PR